spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewritePositionDeleteFilesProcedure.java

```diff
@@ -130,6 +141,42 @@ public void testExpireDeleteFilesNoOption() throws Exception {
         output);
   }
 
+  @Test
+  public void testExpireDeleteFilesFilter() throws Exception {
+    createTable(true);
+
+    sql("DELETE FROM %s WHERE data='a'", tableName);
+    sql("DELETE FROM %s WHERE data='c'", tableName);
+    sql("DELETE FROM %s WHERE data='e'", tableName);
```

Comment:
if we delete `a, b, e` instead, does it imply we will generate 3 delete files first and compaction with where clause `id IN (1, 2)` would remove 2 and add 1 back?

Reply:
Yea its a bit of a lame test, I actually just make one delete file in each partition (3 partitions, 3 files), and then rewrite all to force 2 out of 3 to be rewritten because I have a partition filter.  The rewrite just puts those 2 back as is, because they cant go across partition boundaries.